File: frontend/types/project.ts
export type ProjectStatus = "active" | "archived";

export type Project = {
  id: string;
  name: string;
  projectNumber: string | null;
  client: string | null;
  location: string | null;
  buildingType: string | null;
  description: string | null;
  status: ProjectStatus;
  createdAt: string;
  updatedAt: string;
  /** Not yet returned by the backend (no archive endpoint exists — ADR pending); optional so real API responses satisfy this type without fabricating a value. */
  archivedAt?: string | null;
};

/** What the Projects List and Workspace Overview actually render. */
export type ProjectSummary = Project & {
  documentCount: number;
  openFindingCount: number;
  lastActivityAt: string | null;
};

File: backend/app/schemas/project.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict
from pydantic.alias_generators import to_camel

from app.models.project import ProjectStatus


class ProjectResponse(BaseModel):
    model_config = ConfigDict(alias_generator=to_camel, populate_by_name=True, from_attributes=True)

    id: str
    name: str
    project_number: str | None
    client: str | None
    location: str | None
    building_type: str | None
    description: str | None
    status: ProjectStatus
    created_at: datetime
    updated_at: datetime
